torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -1156,22 +1157,41 @@ struct PythonPrintImpl {
         printOpName(stmt, node->kind());
         const FunctionSchema& schema = node->schema();
         stmt << "(";
-        for (size_t i = 0; i < node->inputs().size(); ++i) {
-          if (i > 0) {
-            stmt << ", ";
-          }
-          auto v = useOf(node->inputs().at(i));
-          // print the kwarg name if it is a kwarg only argument.
-          if (i < schema.arguments().size()) {
-            auto arg = schema.arguments().at(i);
+        // only ignore arguments when we see non vararg schemas
+        if (!schema.is_vararg()) {
+          // calculate how many args are specified.
+          // see (https://github.com/pytorch/pytorch/pull/56079) for more
+          // details.
+          int necessary_args =
+              calculateNecessaryArgs(schema.arguments(), node->inputs());
+          for (int index = 0; index < necessary_args; ++index) {
+            if (index > 0)
+              stmt << ", ";
+            auto v = useOf(node->inputs().at(index));
+            auto arg = schema.arguments().at(index);
             if (arg.kwarg_only()) {
               stmt << arg.name() << "=";
             }
-          } else {
-            // vararg functions like format can have extra arguments
-            AT_ASSERT(schema.is_vararg());
+            stmt << *v;
```

Comment:
This `for` loop looks almost identical to the loop used in `is_vararg()` branch, other than the loops have different upper bound. Could we consider consolidating them into one?

Reply:
I thought it would help with readability. If we consolidate them together, I feel like it would be too complicated to understand.